File: LeetCodeChallenges/CodeWithClass/Day02/C++/trial.cpp
#include <vector>
using namespace std;

class Solution {
    public:
        vector<int> findDisappearedNumbers(vector<int>& nums) {
            int n = nums.size();
            vector<bool> ref(n, false);
            for (int num : nums) {
                ref[num - 1] = true;
            }

            vector<int> result;
            for (int i = 0; i < n; i++) {
                if (!ref[i]) {
                    result.push_back(i + 1);
                }
            }
            return result;
        }
};

int main() {
    Solution sol;
    vector<int> nums = {4, 3, 2, 7, 8, 2, 3, 1};
    vector<int> result = sol.findDisappearedNumbers(nums);
    return 0;
}

File: LeetCodeChallenges/CodeWithClass/Day02/python/trial.py
nums = [4, 3, 2, 7, 8, 2, 3, 1]

for num in nums:
    index = abs(num) - 1
    print(f"num: {num}, index: {index}, nums[index]: {nums[index]}")

    nums[index] = -abs(nums[index])
    print(f"nums[index]: {nums[index]}")
    print()

print("nums[i] > 0")
print([i + 1 for i in range(len(nums)) if nums[i] > 0])
print("nums[i] < 0")
print([i + 1 for i in range(len(nums)) if nums[i] < 0])
print()